sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -239,6 +243,19 @@ public <T> T insert(String containerName, T objectToSave, PartitionKey partition
         return toDomainObject(domainType, response.getItem());
     }
 
+    public <S extends T, T> Iterable<S> insertAll(String containerName, Class<T> domainType, Flux<CosmosItemOperation> cosmosItemOperationFlux) {
```

Comment:
This is where I am having trouble, I can't get the return type to line up with the method return type. It is cast as Iterable<S> right now so that it complies but it does not wokr.

Reply:
Why don't we use the following directly?

```java
public <T> Iterable<T> insertAll(String containerName, Class<T> domainType, Flux<CosmosItemOperation> cosmosItemOperationFlux) {
```

Reply:
@saragluna it is called by SimpleCosmosRepository.saveAll() which doesn't allow this format to be returned to it.